LIPS/lip-25.md

```diff
@@ -0,0 +1,1983 @@
+) external onlyRole(REPORT_EXITED_VALIDATORS_ROLE) {
+    _checkValidatorsByNodeOperatorReportData(_nodeOperatorIds, _stuckValidatorsCounts);
+    _getIStakingModuleById(_stakingModuleId).updateStuckValidatorsCount(_nodeOperatorIds, _stuckValidatorsCounts);
+}
+
+/// @notice Returns node operator digest for each node operator registered in the given staking module
+/// @param _stakingModuleId id of the staking module to return data for
+/// @dev WARNING: This method is not supposed to be used for onchain calls due to high gas costs
+///     for data aggregation
+function getAllNodeOperatorDigests(uint256 _stakingModuleId) external view returns (NodeOperatorDigest[] memory) {
+    return
+        getNodeOperatorDigests(_stakingModuleId, 0, _getIStakingModuleById(_stakingModuleId).getNodeOperatorsCount());
+}
+
+/// @notice Returns node operator digest for passed node operator ids in the given staking module
+/// @param _stakingModuleId id of the staking module where node operators registered
+/// @param _offset node operators offset starting with 0
+/// @param _limit the max number of node operators to return
+/// @dev WARNING: This method is not supposed to be used for onchain calls due to high gas costs
+///     for data aggregation
+function getNodeOperatorDigests(
+    uint256 _stakingModuleId,
+    uint256 _offset,
+    uint256 _limit
+) public view returns (NodeOperatorDigest[] memory) {
+    return
+        getNodeOperatorDigests(
+            _stakingModuleId,
+            _getIStakingModuleById(_stakingModuleId).getNodeOperatorIds(_offset, _limit)
+        );
+}
+
+/// @notice Returns node operator digest for a slice of node operators registered in the given
+///     staking module
+/// @param _stakingModuleId id of the staking module where node operators registered
+/// @param _nodeOperatorIds ids of the node operators to return data for
+/// @dev WARNING: This method is not supposed to be used for onchain calls due to high gas costs
+///     for data aggregation
+function getNodeOperatorDigests(
+    uint256 _stakingModuleId,
+    uint256[] memory _nodeOperatorIds
+) public view returns (NodeOperatorDigest[] memory digests) {
+    IStakingModule stakingModule = _getIStakingModuleById(_stakingModuleId);
+    digests = new NodeOperatorDigest[](_nodeOperatorIds.length);
+    for (uint256 i = 0; i < _nodeOperatorIds.length; ++i) {
+        digests[i] = NodeOperatorDigest({
+            id: _nodeOperatorIds[i],
+            isActive: stakingModule.getNodeOperatorIsActive(_nodeOperatorIds[i]),
+            summary: getNodeOperatorSummary(_stakingModuleId, _nodeOperatorIds[i])
+        });
+    }
+}
+
+function getStakingModuleNonce(uint256 _stakingModuleId) external view returns (uint256) {
+    return _getIStakingModuleById(_stakingModuleId).getNonce();
+}
+```
+
+#### 2.7. Excluded methods for module pausing
+
+Methods, events and constants related to Staking Module pausing logic should be removed from the `StakingRouter` contract:
```

Comment:
And it's proposed to have pause for all deposits on DSM side and unvetting. The motivation is that in the current design if one permisionless operator frontruns deposits than DSM pauses the whole module. We need to change this reaction to a more specific one, from a module to a specific operator.

Reply:
OK, I've added more details about the purpose of the new pause to the 2.6 section. Also, I've added the link to the new DSM spec, where much more details about this logic are provided.